lib/result/value_isnt_result_error.ex

```diff
@@ -0,0 +1,9 @@
+defmodule Result.ValueIsntResultError do
```

Comment:
Ten nazev modulu neodpovida tomu raisu, tj. ta vyjimka nastane kdyz result nebude `:ok` nebo `:error`.  Navic se mi ted nazev zda nejak kostrbaty :-( Neslo by treba `Result.Exception` nebo `Result.CheckType` ?

Reply:
`Exception` urcite ne a je to chyba, takze `CheckType` jedine jako `MishmashTypeError`.